lib/Runtime/Types/DynamicObject.cpp

```diff
@@ -895,6 +895,13 @@ namespace Js
 
     bool DynamicObject::TryCopy(DynamicObject* from)
     {
+#if ENABLE_TTD
+        if (from->GetScriptContext()->ShouldPerformRecordOrReplayAction())
```

Comment:
Does this essentially disable type sharing during record/replay scenarios?

Reply:
It should disable the optimization around object assignment during record and replay. Other type sharing should be enabled.